lib/swaps/Swaps.ts

```diff
@@ -233,10 +290,89 @@ class Swaps extends EventEmitter {
       createTime: Date.now(),
     };
 
+    // add the deal. Going forward we can "record" errors related to this deal.
     this.addDeal(deal);
 
+    // verify LND setup. Make sure we are connected to BTC and LTC and that
+    // the peer is also connected to these networks.
+    const err = this.verifyLNDSetup(deal, peer);
+    if (err) {
+      this.setDealState(deal, SwapDealState.Error, err);
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+      return;
+    }
+
+    const heldAmount = this.checkAndHoldAmount(deal);
+    assert(heldAmount === 0 || heldAmount === deal.takerAmount,
+        'Partial amount is not yet supported');
+    if (heldAmount === 0) {
+      this.setDealState(deal, SwapDealState.Error, 'Can not swap. No capacity is available');
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+      return;
+    }
+
+    let lndclient: LndClient;
+    switch (deal.takerCurrency) {
+      case 'BTC':
+        lndclient = this.lndBtcClient;
+        break;
+      case 'LTC':
+        lndclient = this.lndLtcClient;
+        break;
+      default:
+        this.setDealState(deal, SwapDealState.Error, 'Can not swap. Unsupported Taker currency.');
+        this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+        return;
+    }
+
+    let height: number;
+    try {
+      const req = new lndrpc.QueryRoutesRequest();
+      req.setAmt(requestBody.takerAmount);
+      req.setFinalCltvDelta(requestBody.takerCltvDelta);
+      req.setNumRoutes(1);
```

Comment:
Do we want to query for more than 1 route in case payment on the first route fails for some reason? Or would this be a future enhancement, and if so maybe add a TODO?